hydromt/components/spatial.py

```diff
@@ -28,27 +29,36 @@
 
 logger = getLogger(__name__)
 
-DEFAULT_REGION_FILE_PATH = "region.geojson"
 
-
-class ModelRegionComponent(ModelComponent):
+class SpatialModelComponent(ModelComponent, ABC):
     """Define the model region."""
 
+    DEFAULT_REGION_FILENAME = "region.geojson"
+    DEFAULT_HYDROGRAPHY_FILENAME = "merit_hydro"
+    DEFAULT_BASIN_INDEX_FILENAME = "merit_hydro_index"
```

Comment:
In other components, the defaults were written outside of the class.
Also hydrography and basin index should not have defaults.